lib/spack/spack/cmd/spec.py

```diff
@@ -78,10 +82,18 @@ def spec(parser, args):
     if args.install_status:
         tree_context = spack.store.db.read_transaction
 
-    if not args.specs:
-        tty.die("spack spec requires at least one spec")
-
-    for spec in spack.cmd.parse_specs(args.specs):
+    # Use command line specified specs, otherwise try to use environment specs.
+    if args.specs:
+        specs = spack.cmd.parse_specs(args.specs)
+    else:
+        env = ev.active_environment()
+        if env:
+            env.concretize()
+            specs = [spec for _, spec in env.concretized_specs()]
+        else:
+            tty.die("spack spec requires at least one spec")
```

Comment:
```suggestion
        env = spack.cmd.require_active_env(cmd_name='spec')
        env.concretize()
        specs = [spec for _, spec in env.concretized_specs()]
```

Reply:
Hm yeah, I thought the typical use case for `spack spec` is to concretize a cli-provided spec, so if the user forgets to list a spec, an error about a missing required environment may be unexpected/confusing?

Reply:
I can make the error message: `spack spec requires at least one spec or an active environment`.

Reply:
```
==> Error: install requires a package argument or active environment
```
is what `spack install` says, so let's mirror that.